hastexo/tasks.py

```diff
@@ -625,6 +626,11 @@ def run(self, **kwargs):
             'status': status,
         }
 
+        # Set the delete_by value based on the delete_age value
+        if not stack.delete_by and status != SUSPEND_FAILED:
```

Comment:
Wait, hold on. This doesn't look right.

First, I don't think it make sense to set `delete_by` only if it isn't already set. The idea is that a stack gets reaped _if it hasn't been resumed in some time,_ so setting the `delete_by` state only on the first suspend, rather than updating it on every suspend, defeats that purpose.

Second, I think we want to set this _even if_ `status == SUSPEND_FAILED` because otherwise a stack might fail to suspend, and subsequently linger forever as a result. 

Or am I missing something?

Reply:
Right, I'll need to trace my thoughts on this, I believe you are right. We should just set the `delete_by` value here I think.